modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -158,17 +181,14 @@ public MetadataMediator(RuntimeExtensionModel extensionModel, RuntimeComponentMo
    * If Component's {@link Content} parameter has a {@link MetadataContentResolver} associated or its Output has a
    * {@link MetadataOutputResolver} associated that can be used to resolve dynamic {@link MetadataType}, then the
    * {@link ComponentMetadataDescriptor} will contain those Dynamic types instead of the static type declaration.
-   * <p>
-   * When neither {@link Content} nor Output have Dynamic types, then invoking this method is the same as invoking
-   * {@link this#getMetadata()}
    *
    * @param context current {@link MetadataContext} that will be used by the {@link MetadataContentResolver} and
-   *        {@link MetadataOutputResolver}
-   * @param key {@link MetadataKey} of the type which's structure has to be resolved, used both for input and output types
+   *                {@link MetadataOutputResolver}
+   * @param key     {@link MetadataKey} of the type which's structure has to be resolved, used both for input and output types
    * @return Successful {@link MetadataResult} if the MetadataTypes are resolved without errors Failure {@link MetadataResult}
-   *         when the Metadata retrieval of any element fails for any reason
+   * when the Metadata retrieval of any element fails for any reason
    */
-  public MetadataResult<ComponentMetadataDescriptor> getMetadata(MetadataContext context, MetadataKey key) {
+  private MetadataResult<ComponentMetadataDescriptor> getMetadata(MetadataContext context, Object key) {
```

Comment:
why? We can always return a valid result, either success or failure, but with a complete ComponentMetadataDescriptor